### DISCUSSION TITLE
How to have global store(s) which are reactive (ideally fine grained)?

### DISCUSSION THREAD

**Discussion by dominictobias-bullish** [lapce/floem#553] [Q&A] [answered]
Hi, I know there is `provide_context` + `use_context` but is it reactive to state changes? Looking at the code it seems not?

I would like to be able to have a `stuct` of variables and I can bind one or more of the variables inside in a view so that it updates when they change. e.g. `const email = useUserStore(state => state.email)` in a typical React store.

I didn't see an example of state from `provide_context` being changed in here or Lapce.

**Comment by jrmoulton** [comment, marked as answer]
If you store a signal with provide context and then update that signal it will be reactive
